### PR TITLE
T04_Add_possibility_to_delete_todo

### PR DESCRIPTION
**Pull request by APaltrow**
Task: T04_Add_possibility_to_delete_todo

-Add "trash" icon right to the existing item
-When "trash" icon is clicked delete item from the list

### CODE REVIEW

File: src/components/ui/Icon/Icon.tsx [hunk 1]
```diff
@@ -5,7 +5,7 @@ import sprites from '@assets/sprites.svg';
 import style from './Icon.module.scss';
 
 interface IconProps {
-  iconName: string;
+  iconName: 'plus' | 'delete' | 'close';
```
Comment (Yuliia143) [inline src/components/ui/Icon/Icon.tsx]: Try to use enums

File: src/redux/reducers/todoReducer.ts [hunk 2]
```diff
@@ -29,6 +29,14 @@ export const todoReducer = (
       return { ...state };
     }
 
+    case TodoActions.DELETE_TODO: {
+      const updatedList = state.todoList.filter(
+        (todo) => todo.id !== action.payload,
+      );
+
+      return { ...state, todoList: [...updatedList] };
```
Comment (Yuliia143) [inline src/redux/reducers/todoReducer.ts]: updatedList is already a new array, why do you need to use spread operator ?

File: src/components/ui/Icon/Icon.tsx (hunk 1 above)
Comment (APaltrow) [reply src/components/ui/Icon/Icon.tsx]: Decomposed iconName to IconsTypes enum. Updated relevant components.

File: src/redux/reducers/todoReducer.ts (hunk 2 above)
Comment (APaltrow) [reply src/redux/reducers/todoReducer.ts]: Changed spredded updatedList to just passing the link for updatedList.

**Review by Yuliia143 — APPROVED** [review]

**Pull request merged**